src/schemas/__tests__/activities.schema.unit.spec.js

```diff
@@ -0,0 +1,293 @@
+import { createActivitySchema } from '../activities.schema.js'
+import { isActivityExists } from '../../services/activities.service.js'
+import { isRiverInternal } from '../../services/rivers.service.js'
+import { isSubmissionExists } from '../../services/submissions.service.js'
+
+jest.mock('../../services/activities.service.js')
+jest.mock('../../services/rivers.service.js')
+jest.mock('../../services/submissions.service.js')
+
+describe('activities.schema.unit', () => {
+  describe('createActivitySchema', () => {
+    const mockCurrentYear = (year) => {
+      jest.useFakeTimers('modern')
+      jest.useFakeTimers().setSystemTime(new Date(`${year}-01-01`))
+    }
+
+    afterEach(() => {
+      jest.useRealTimers()
```

Comment:
I think you also need to do `jest.resetAllMocks()`, as you're setting resolved values in a lot of tests but not clearing them before they're re-used, so some tests rely on setup done in other tests. After this you'll probably have some test failures and have to make sure each test sets the mocks up as it needs them to be.